presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -220,6 +220,16 @@ else if (controlResult.getState() != State.SUCCESS) {
             }
         }
 
+        // Re-run the test query to confirm that the results don't match, in case the underlying data changed, but was cached on the test tier.
+        QueryResult results = executeQuery(testGateway, testUsername, testPassword, queryPair.getTest(), testTimeout, sessionProperties);
+        if (results.getState() != State.SUCCESS) {
+            return false;
+        }
+        testResult = results;
+        if (resultsMatch(controlResult, testResult)) {
+            return true;
+        }
+
```

Comment:
Ya, although that's also the point of this change, since caching is making them non-deterministic. Do you think it would be better to re-run the query several times, say 3, like is done for the control to look for non-determinism?